File: app/src/main/java/soup/movie/ui/now/NowViewState.java
package soup.movie.ui.now;

import android.support.annotation.NonNull;

import java.util.List;

import soup.movie.data.soup.model.Movie;
import soup.movie.ui.BaseViewState;

interface NowViewState extends BaseViewState {

    class LoadingState implements NowViewState {

        LoadingState() {
        }

        @Override
        public String toString() {
            return "LoadingState{}";
        }
    }

    class DoneState implements NowViewState {

        private final String mTitle;
        private final List<Movie> mMovies;

        DoneState(@NonNull String title, @NonNull List<Movie> movies) {
            mTitle = title;
            mMovies = movies;
        }

        @NonNull
        public String getTitle() {
            return mTitle;
        }

        @NonNull
        public List<Movie> getMovies() {
            return mMovies;
        }

        @Override
        public String toString() {
            return "DoneState{" +
                    "mTitle='" + mTitle + '\'' +
                    ", mMovies=" + mMovies +
                    '}';
        }
    }
}

File: app/src/main/java/soup/movie/ui/main/now/NowViewState.kt
package soup.movie.ui.main.now

import androidx.annotation.Keep
import soup.movie.data.model.Movie

sealed class NowViewState {

    @Keep
    object LoadingState : NowViewState() {

        override fun toString(): String = javaClass.simpleName
    }

    @Keep
    object ErrorState : NowViewState() {

        override fun toString(): String = javaClass.simpleName
    }

    @Keep
    data class DoneState(val movies: List<Movie>) : NowViewState()
}
